src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -416,28 +415,135 @@ def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):  # py
+    from .repository import get_access_credentials
+    from ._docker_utils import request_data_from_registry
+    from azure.cli.core.azclierror import ResourceNotFoundError
+
+    registry, _ = get_registry_by_name(cmd.cli_ctx, registry_name)
+    login_server = registry.login_server.rstrip('/')
+    path = '/v2/' + repository_name + '/manifests/' + image
+
+    # Get the access credentials for the registry
+    registry_name, username, password = get_access_credentials(
+        cmd,
+        registry_name=registry_name,
+        tenant_suffix=None,
+        username=None,
+        password=None,
+        repository=repository_name,
+        permission='pull')
+    
+    try:
+        request_data_from_registry(
+            http_method='get',
+            login_server=login_server,
+            path=path,
+            username=username,
+            password=password,
+            json_payload=None,
+            file_payload=None,
+            manifest_headers={'Accept': 'application/vnd.docker.distribution.manifest.v2+json'},
+            params=image)
+    except CLIError as e:
+        # Error is due to the image tag not being found on manifest
+        if "manifest tagged by" in str(e):
+            raise ResourceNotFoundError(IMAGE_INPUT_ERROR_MESSAGE)
+        # Error is due to the repository not being found
+        else:
+            raise ResourceNotFoundError(REPOSITORY_INPUT_ERROR_MESSAGE)
+
+# Validate if specific blob can be pulled
+def _check_blob_exists(cmd,
+                       registry_name,
+                       repository_name,
+                       image):
+    
+    import requests
+    from .repository import get_access_credentials
+
+    registry, _ = get_registry_by_name(cmd.cli_ctx, registry_name)
+    login_server = registry.login_server.rstrip('/')
+
+    # Get the access credentials for the registry
+    registry_name, username, password = get_access_credentials(
+        cmd,
+        registry_name=registry_name,
+        tenant_suffix=None,
+        username=None,
+        password=None,
+        repository=repository_name,
+        permission='pull')
+    
+    # Get manifest
+    # GET {url}/v2/{name}/manifests/{reference}
+    manifest_url = 'https://{}/v2/{}/manifests/{}'.format(login_server, repository_name, image)
+    
+    response = requests.get(
+        manifest_url,
+        headers={'Accept': 'application/vnd.oci.artifact.manifest.v1+json'
+            ', application/vnd.cncf.oras.artifact.manifest.v1+json'
+            ', application/vnd.oci.image.manifest.v1+json'
+            ', application/vnd.oci.image.index.v1+json'
+            ', application/vnd.docker.distribution.manifest.v2+json'
+            ', application/vnd.docker.distribution.manifest.list.v2+json'},
+        auth=(username, password))
+    
+    manifest = response.json()
+
+    # Get the digest of the smallest blob for performance purposes
+    def get_smallest_blob_digest(manifest):
```

Comment:
why are we defining a function inside of another function for it to only be used once? Either define it as a helper function the top level so it can be reused by other functions or just write the code directly in this function